pyproject.toml

```diff
@@ -1,3 +1,27 @@
+# Specification of pyproject.toml:
+# https://packaging.python.org/en/latest/guides/writing-pyproject-toml/
+
+# Should pyproject.toml be added?
+# A pyproject.toml file is strongly recommended. The presence of a pyproject.toml 
+# file itself does not bring much. What is actually strongly recommended is the 
+# [build-system] table in pyproject.toml.
+# https://packaging.python.org/en/latest/guides/modernize-setup-py-project/#should-pyproject-toml-be-added
+
+# Choosing a build backend:
+# https://packaging.python.org/en/latest/tutorials/packaging-projects/#choosing-a-build-backend
+[build-system]
+# Both of these values will be provided by the documentation for your build backend, 
+# or generated by its command line interface.
+# Usually, you’ll just copy what your build backend’s documentation suggests.
+# There should be no need for you to customize these settings.
+#
+# A list of packages that are needed to build your package:
+requires = ["setuptools"]  # REQUIRED if [build-system] table is used
+# The name of the Python object that frontends will use to perform the build.
+# If not defined, then the fall-back behaviour is to use setuptools.
```

Comment:
Most people around here default to the US spellings, not UK. Plus this is not fully accurate.
```suggestion
# If not defined, then the fall-back behavior is to use
# the setuptools' outdated method of operation implemented as
# a `setuptools.build_meta:__legacy__` interface on the
# build backend level.
```

Reply:
thanks!

just a small question:

On the [other page](https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-build-system-dependencies-the-build-system-table), I've seen:

> If the table is specified but is missing required fields then the tool should consider it an error

So the other build backend chooses what to do? - then it's not `setuptools` at all in this case?

Reply:
The `[build-system]` table is consumed by build *frontends*. `setuptools` is a *backend* and wouldn't normally access it. The frontend creates an isolated virtualenv for building the project and pre-provisions it with build dependencies. For that, it consults the `requires` field in order to pre-install some packages into the temporary build environment (there are actually 3 other PEP 517 / PEP 660 hooks that can be consulted for *additional build deps*). Only when the ephemeral build environment is ready, the frontends go ahead and call the hooks defined in the backend that is specified or defaulted to.